settings-selected-project.txt
atmosphere	25/06/20	fadfb06af0a42cdc3ec5a364b7cf940f344629d6	27/04/10	7529a8bdebf7c6b2f1e332a9b3e5664ec677bd76
checkstyle	01/05/20	164a755af951cf0fd459d70873e1c199210d9d8b	22/06/01	5f53a2a380e038e791af244b42f245b813dbc379
spring-framework	08/06/20	1984cfe9d79e8a34955a393ef9d71a2a3d11a87c	10/07/08	8119659fb1e4ae6fabe8897c42ba7629fda07b21
mockito	19/08/18	077562ea54f1fa87ff8dd233c3060ddbf0f1ce26	15/11/07	7cf6470d88491fe472e28493b50eba8a6fbf0433
jetty	21/08/18	fc5dd874f3deda71e6cd42af994a5af5cb6be4af	16/03/09	bc1e0bd10201d8a14f20a81e3b93076af6408fe4
pmd	27/01/20	d115ca59f73490c771abfe754414722512b4ac9b	21/06/02	f2138120f1d199906541da8948e4303c00ccf1ef
jna	19/07/20	b8443b80351ca3f0cdefe2752c7209abf0f5d819	01/10/98	9813273b07fd082da573b3b6bfb8d23809b59eea
jgit	09/06/20	855842af1934dd02f2bf20b4e723d63ba5786ba0	29/09/09	1a6964c8274c50f0253db75f010d78ef0e739343
junit4	10/04/20	50a285d3ce69b4556ac46d8633f6beb4527b4679	03/12/00	b6a0693454ac8ded32b3a1ea7c859c5a840169dc
junit5	06/06/20	be2aa2497a75dedceb37bc3297276e230fb4f293	04/06/15	92c14e9cec7862254d06019043d081d4e8c64789
hbase	13/07/20	3bd54217a04e30db09e3261626687fc8779061b3	03/04/07	948d0cdec4fb35d70404b9b6e27743616400656d
cassandra	12/06/20	7cdad3ce61664cf2ffac75b6e537c910e573754c	02/03/09	1f91e99223b0d1b7ed8390400d4a06ac08e4aa85
wildfly	30/06/20	f21f5db2c019bf96030e5ae388bc3248e7ee6552	28/05/10	033fc94f0829b291e7a458e3b776217924d1b445
intellij-community	31/01/20	cdf2efe8b88b87149784eae02241001ea60a8c29	05/11/04	3dbff843ac8df3aa167ddf0a52f5869086daa45a
weka	24/04/19	a22631f8ba399bdfd92f9dd2f44a9d80544ffd6d	20/04/99	c0dc68970a412a97058ace5d7e0cc93dd399a527
javaparser	24/04/20	8f25c40918feaf5f66af5fae5b23d271edefbdfc	30/10/11	7b2741cfb627157580719dd43c46b716bde4267e
sonarqube	22/06/20	6b806ebf8e189b2abf7d29c964abf09a3a3beb8c	06/09/10	aeadc1f9129274949daaa57738c7c4550bdfbc7b
hazelcast	07/07/20	a59ad427a0afe6e0beb1a6adf69b7c6d84c10	15/12/08	e29759246fc919efca6862e41652a2b4c502c5b4
hadoop	27/05/20	4c5cd751e3911e350c7437dcb28c0ed67735f635s	19/05/09	5128a9a453d64bfe1ed978cf9ffed27985eeef36
wicket	10/06/20	e3f3704888f08e32040f21cb41b116cce43acfce	21/09/04	f92c2a5ad87198c9e446b271a713942bb89d89d2
argouml	22/05/20	fcbe6c6eb764525a7509d8bc8bdfa19aea551c9f	26/01/98	3fc0287e1c4b873e13d1f928e4c4f4413a5cac69
hector	30/07/15	a302e68ca8d91b45d332e8c9afd7d98030b54de1	10/02/10	a309ee7fd0626cc23032e9794e494a1e7241a446
xerces2-j	06/04/19	cf0c517a41b31b0242b96ab1af9627a3ab07fcd2	09/11/99	045efdd5ecc8773b3bf3dccbda6338f6d5076a93
openmrs-core	18/07/20	c5928a553593ba2f798fad220b7627519ccbdf71	02/05/06	b57cd757597e9146f606abe8745ed08548e1e979
hibernate-search	24/07/18	5b778035965d7588ad1d1ae522c4bafebd3a0e16	15/08/07	fced424eefdd628fc096ccea5689dcff1cbcc78a
okhttp	19/07/18	5224f3045ba9b171fce521777edf389f9206173c	25/05/11	6594b2a8999dce7787010652d6767cf0af1bd491
mongo-java-driver	11/07/20	8ab109b31eba9ea99ea459a3ce0aad988442985e	06/01/09	f13afc7902839eff968d5abd88f3d9a40c90f7a1
lombok	10/07/20	4fdcdd126a4b5ce365b97eb498536d04caee3b53	08/06/09	1677c4a52a0aea1b955f7c2c7d096903d4a8c5ce
cucumber-jvm	04/07/20	b57b924bd19e70f1dac2d018dc9b2e58ea12de9f	28/01/09	c654ee66bd637d344934bf48d6a6b3ac030223f6
Essentials	15/06/15	d36d80933f8f672cd8bb0f210adc23aac10850ea	19/03/11	e1dc6cf03069011eac824884f4e88bd33d689468
titan	03/12/15	ee226e52415b8bf43b700afac75fa5b9767993a5	26/02/12	39dbfe175e9d07a699987871ab301034d1190912
facebook-android-sdk	04/12/2020	fb1b91a32c8ac21ec91800748a24e3c20b21ac0e	10/05/10	b119d7cb840437257d5f584f2e8ebf5f77ec6f55
netty	25/06/20	662e0b4b81833a835b290e60982c9de3a1878656	08/08/08	aef2ab453a8b11984c4fe64ba27612a1308ee490
flink	01/05/20	261e72119b69c4fc3e22d9bcdec50f6ca2fdc2e9	15/12/10	3b88e30924268799c96317fe1bf9f5b9c6bf6f80
jclouds	04/04/20	7af4d8e8f19c479b0a8f35e06dd68418a8367b3e	11/04/09	8efb9ce0dc1c00e99be9c8d52585d1681a8f9524
hibernate-orm	30/04/20	2c12ca401dcc55b296522a54135169c6e5fe57e4	29/06/07	41e94b3cf8192f0f3b4d2948f26760ac412745a0
vraptor4	26/09/18	593ce9ad60f9d38c360881b2132417c56b8cc093	02/05/13	e1765683a0f781a960c527ce281165648a877aaa
commons-io	25/12/19	11f0abe7a3fb6954b2985ca4ab0697b2fb489e84	25/01/02	9423d7dc927d7d44980b08b59b2a127f8f26ae99
docx4j	08/07/20	36c3787d1d858728dc48ffe8af2a59e6bdf8230c	29/10/07	db29775dcc8a68bf50846791daa904974a62059d
lucene-solr	01/02/20	b457c2ee2ea0180f3a11c5d0b4d443083316aa31	11/09/01	a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85
ant	03/06/20	1ce1cc2317cb2b2773ab6b765f1a79ccd4806926	13/01/00	96d856e389bb88f6d0815cd8f48643e5ce8b7957
spring-boot	01/02/20	199cea206ffe316f60a91122207bcaea613878c9	21/10/12	80b151e2b37a66a8fe7eb2d5bac2f7aa766d4ead
presto	04/12/2020	bb20eba263685c6ae7b8ee71e405746eed579431	08/08/12	d0eb103a660125a53660a7ba2c95e9b8a4d8feb6
voldemort	30/08/17	a7dbdea58032021361680faacf2782cf981c5332	17/04/11	c38773e0f545a1b8443464c73f54e7855c048dd2
guava	30/06/20	e352076da65aa3afb5dc59cfd3c1f41574b5ebeb	18/06/09	db3929bd79ef57f8da40515b663f40200a941fa8
eclipseJdt	19/06/20	475591df71dc3e97c1ebea9d583779b10c2f1505	05/06/01	be6c0d208933ac936a6ccb6c66b03d3da13e3796
elasticsearch	31/01/20	92be385f8ea3d39cc42155417d208ba82423d983	08/02/10	ec72ca8b7a115f9b2eea3c76c518062b99a1d015
commons-lang	20/04/20	f6923510352fc3fbfad68bc6c5ac5258a34671b7	19/07/02	750a21e86441b247d85c95e59bf17e596a69d626
RxJava	06/07/20	880eed2ac0dc1e3fc234bb652f5e177ce643093c	18/03/12	697fd66aae9beed107e13f49a741455f1d9d8dd9


Bug-prediction/src/main/java/main/Settings.java
package main;

public class Settings {

  public  static String BASE_PATH="/home/shaiful/research/method-level-bug-prediction/mydata/";
  public  static String PROJECTS=BASE_PATH+"info/settings-project-with-first-date.txt";
 // public final static String BASE_DIR_SOURCE_CODE ="/home/shaiful/research/codeshovel/codeshovel-projects/";
  public  static String JSON_HISTORY_DIR ="/home/student/kidwaih1/Documents/Tech-Debt/source-methods-49-projects/source-methods/";
  public  static String RESULT_DIR=BASE_PATH+"data/";
// for test projects, filtering would be false
  public static boolean FILTER_OUT_TEST_METHODS = true;
//  public static String HALSTED_PATH = "/home/shaiful/research/code-metrics/software-evolution/jars/halstead.jar";

  // THE FOLLOWING VALUES DOES NOT MATTER AS EVERYTHING WILL COME FROM FILE
  // I.E., SETTINGS.PROJECTS

  public static String PROJECT="";
  public static String CURRENTDATE =""; //dd/mm/yy
  public static String PROJECT_FIRST_DATE = "";

//this will change based on the project last commit where the methods' history started
// is not needed when working from file (S)

// mytest project for accuracy
//  public static String CURRENTDATE ="29/06/20"; //dd/mm/yy

//checkstyle
//  public final static String CURRENTDATE ="01/05/20"; //dd/mm/yy

//hadoop
//  public final static String CURRENTDATE ="27/05/2020";

//commons-io
// public final static String CURRENTDATE ="25/12/19"; //dd/mm/yy

//lucene-solr
//public final static String CURRENTDATE ="01/02/20"; //dd/mm/yy

//mockito
//public final static String CURRENTDATE ="19/08/18"; //dd/mm/yy

//hibernate-search
// public final static String CURRENTDATE ="24/07/18"; //dd/mm/yy

// jetty (had to follow codeshovel's commit)
//Fri Aug 17 16:57:28 2018 -0500

// flink
//    Fri May 1 01:12:12 2020 +0800


}
